conda/resolve.py

```diff
@@ -896,11 +923,33 @@ def mysat(specs, add_if=False):
             solution, obj7 = C.minimize(eq_optional_c, solution)
             log.debug('Package removal metric: %d', obj7)
 
-            # Requested packages: maximize versions
+            if context.alt_features:
+                # Requested packages (split by features): maximize versions
+                eq_req_c_f, eq_req_v_f, eq_req_b_f, eq_req_t_f = (
+                    r2.generate_version_metrics(C, specr, split_by_features=True))
+                solution, obj3a_f = C.minimize(eq_req_c_f, solution)
+                solution, obj3_f = C.minimize(eq_req_v_f, solution)
+                log.debug('Initial package channel/version metric (feature split): %d/%d',
+                          obj3a_f, obj3_f)
+                # Requested packages (split by features): maximize builds
+                solution, obj4_f = C.minimize(eq_req_b_f, solution)
+                log.debug('Initial package build metric (feature split): %d', obj4_f)
+
+                # Remaining packages (split by features): maximize versions, then builds
+                eq_c_f, eq_v_f, eq_b_f, eq_t_f = (
+                    r2.generate_version_metrics(C, speca, split_by_features=True))
+                solution, obj5a_f = C.minimize(eq_c_f, solution)
+                solution, obj5_f = C.minimize(eq_v_f, solution)
+                solution, obj6_f = C.minimize(eq_b_f, solution)
+                log.debug('Additional package channel/version/build metrics (feature split): '
+                          '%d/%d/%d', obj5a_f, obj5_f, obj6_f)
+
             eq_req_c, eq_req_v, eq_req_b, eq_req_t = r2.generate_version_metrics(C, specr)
```

Comment:
This seems redundant if `context.alt_features` is True, so shouldn't it be moved inside the if statement that follows? Yes, you need `eq_req_t` at the bottom, but that should be equivalent to `eq_req_t_ft.

Reply:
Yeah,  I added the `if not` only in a later commit https://github.com/conda/conda/pull/6296/commits/9a7857b2b2f6d4f7e9cb8257d2b9538df5706d3a and think I did this inconsistently.
To be certain whether we can use the `eq_req` from the `split_by_features=True` part:
Picture the (unlikely) scenario where two builds have equal channel, version, build number, dependencies but only differ in required features and timestamp (and all other measures in `solve` turn out equal too). With `split_by_features=True` the distinction on features does not lead to a comparison/distinction, i.e., some `+1` in `generate_version_metrics`. Then the `eqt` from `split_by_features=True` could be equal (given certain preconditions), i.e., not act as a tiebreaker.
That's a very, very obscure case (and even more obscure than the need for timestamps themselves) -- but still possible, I guess. What do you think?

Reply:
Any new comments on this one?